tests/storage/hotplug.go

```diff
@@ -0,0 +1,402 @@
+				Expect(volumeStatus.Target).ToNot(BeEmpty())
+				res = append(res, fmt.Sprintf("/dev/%s", volumeStatus.Target))
+			}
+		}
+		return res
+	}
+
+	Context("Offline VM", func() {
+		var (
+			vm *kubevirtv1.VirtualMachine
+		)
+		BeforeEach(func() {
+			By("Creating VirtualMachine")
+			template, _ := newVirtualMachineInstanceWithContainerDisk()
+			vm = createVirtualMachine(false, template)
+		})
+
+		AfterEach(func() {
+			err := deleteVirtualMachine(vm)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("Should add a volume on an offline VM", func() {
+			By("Adding test volumes")
+			addPVCVolumeVM(vm, "some-new-volume1", "madeup", "scsi")
+			addPVCVolumeVM(vm, "some-new-volume2", "madeup", "scsi")
+			By("Verifying the volumes have been added to the template spec")
+			verifyVolumeAndDiskVMAdded(vm, "some-new-volume1", "some-new-volume2")
+			By("Removing new volumes from VM")
+			removeVolumeVM(vm, "some-new-volume1")
+			removeVolumeVM(vm, "some-new-volume2")
+
+			verifyVolumeAndDiskVMRemoved(vm, "some-new-volume1", "some-new-volume2")
+		})
+	})
+
+	Context("Online VM", func() {
+		var (
+			vm *kubevirtv1.VirtualMachine
+			dv *cdiv1.DataVolume
+		)
+
+		BeforeEach(func() {
+			By("Creating VirtualMachine")
+			template := tests.NewRandomFedoraVMIWitGuestAgent()
+			vm = createVirtualMachine(true, template)
+			Eventually(func() bool {
+				vm, err := virtClient.VirtualMachine(tests.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vm.Status.Ready
+			}, 300*time.Second, 1*time.Second).Should(BeTrue())
+
+			By("Creating DataVolume")
+			dv = tests.NewRandomBlankDataVolume(tests.NamespaceTestDefault, corev1.ReadWriteOnce)
+			_, err := virtClient.CdiClient().CdiV1alpha1().DataVolumes(dv.Namespace).Create(dv)
+			Expect(err).To(BeNil())
+			tests.WaitForSuccessfulDataVolumeImport(dv, 240)
+		})
+
+		AfterEach(func() {
+			By("Deleting the virtual machine")
+			Expect(virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &metav1.DeleteOptions{})).To(Succeed())
+			By("Waiting for VMI to be removed")
+			Eventually(func() int {
+				vmis, err := virtClient.VirtualMachineInstance(vm.Namespace).List(&metav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return len(vmis.Items)
+			}, 300*time.Second, 2*time.Second).Should(BeZero(), "The VirtualMachineInstance did not disappear")
+
+			By("Deleting the DataVolume")
+			ExpectWithOffset(1, virtClient.CdiClient().CdiV1alpha1().DataVolumes(dv.Namespace).Delete(dv.Name, &metav1.DeleteOptions{})).To(Succeed())
+		})
+
+		table.DescribeTable("should add/remove volume", func(volumeName, bus string, addVolumeFunc func(vm *kubevirtv1.VirtualMachine, volumeName, claimName, bus string), waitToStart bool) {
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			if waitToStart {
+				tests.WaitForSuccessfulVMIStartWithTimeout(vmi, 240)
+			}
```

Comment:
The waitToStart is a parameter to the table. And I forgot to flip the bottom two. But you are right, I will rework the order, so that the attachments are passed during the boot of the VM, instead of after the console.